# Netflix Content Type Prediction

This project uses machine learning to predict whether a Netflix title is a Movie or a TV Show.

## Project Workflow

1. Load the dataset
2. Select relevant features
3. Encode categorical variables
4. Split the data into training and testing sets
5. Train classification models
6. Evaluate model performance
7. Compare model accuracy

In [8]:
import pandas as pd
df = pd.read_csv("Dataset.csv")
df.head()

,show_id,type,title,director,country,date_added,release_year,rating,duration,listed_in
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,United States,9/25/2021,2020,PG-13,90 min,Documentaries
1,s3,TV Show,Ganglands,Julien Leclercq,France,9/24/2021,2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act..."
2,s6,TV Show,Midnight Mass,Mike Flanagan,United States,9/24/2021,2021,TV-MA,1 Season,"TV Dramas, TV Horror, TV Mysteries"
3,s14,Movie,Confessions of an Invisible Girl,Bruno Garotti,Brazil,9/22/2021,2021,TV-PG,91 min,"Children & Family Movies, Comedies"
4,s8,Movie,Sankofa,Haile Gerima,United States,9/24/2021,1993,TV-MA,125 min,"Dramas, Independent Movies, International Movies"


In [9]:
print("Dataset Shape:", df.shape)

print("\nColumns:")
print(df.columns)

print("\nMissing Values:")
print(df.isnull().sum())

print("\nTarget Values:")
print(df["type"].value_counts())

Dataset Shape: (8790, 10)

Columns:
Index(['show_id', 'type', 'title', 'director', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in'],
      dtype='str')

Missing Values:
show_id         0
type            0
title           0
director        0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
dtype: int64

Target Values:
type
Movie      6126
TV Show    2664
Name: count, dtype: int64


In [10]:
features = ['release_year', 'rating', 'country', 'listed_in']

X = df[features]
y = df['type']

X.head()

,release_year,rating,country,listed_in
0,2020,PG-13,United States,Documentaries
1,2021,TV-MA,France,"Crime TV Shows, International TV Shows, TV Act..."
2,2021,TV-MA,United States,"TV Dramas, TV Horror, TV Mysteries"
3,2021,TV-PG,Brazil,"Children & Family Movies, Comedies"
4,1993,TV-MA,United States,"Dramas, Independent Movies, International Movies"


In [11]:
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder

categorical_features = ['rating', 'country', 'listed_in']
numeric_features = ['release_year']

preprocessor = ColumnTransformer(
    transformers=[
        ('cat', OneHotEncoder(handle_unknown='ignore'), categorical_features),
        ('num', 'passthrough', numeric_features)
    ]
)

X = preprocessor.fit_transform(X)

print("Encoded data shape:", X.shape)

Encoded data shape: (8790, 614)


In [12]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

print("Training data:", X_train.shape)
print("Testing data:", X_test.shape)

Training data: (7032, 614)
Testing data: (1758, 614)


In [13]:
from sklearn.linear_model import LogisticRegression
model = LogisticRegression(max_iter=1000)
model.fit(X_train, y_train)
y_pred = model.predict(X_test)

print("Model training completed")

Model training completed


C:\Users\Fahd\AppData\Roaming\Python\Python314\site-packages\sklearn\linear_model\_logistic.py:599: ConvergenceWarning: lbfgs failed to converge after 1000 iteration(s) (status=1):
STOP: TOTAL NO. OF ITERATIONS REACHED LIMIT

Increase the number of iterations to improve the convergence (max_iter=1000).
You might also want to scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [14]:
from sklearn.metrics import accuracy_score, confusion_matrix

accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", round(accuracy * 100, 2), "%")
print("\nConfusion Matrix:")
print(confusion_matrix(y_test, y_pred))

Accuracy: 99.37 %

Confusion Matrix:
[[1222    3]
 [   8  525]]


In [15]:
from sklearn.ensemble import RandomForestClassifier

rf_model = RandomForestClassifier(
    n_estimators=100,
    random_state=42
)
rf_model.fit(X_train, y_train)
rf_pred = rf_model.predict(X_test)

print("Random Forest training completed")

Random Forest training completed


In [16]:
rf_accuracy = accuracy_score(y_test, rf_pred)

print("Logistic Regression Accuracy:", round(accuracy * 100, 2), "%")
print("Random Forest Accuracy:", round(rf_accuracy * 100, 2), "%")

Logistic Regression Accuracy: 99.37 %
Random Forest Accuracy: 97.1 %


In [17]:
rf_cm = confusion_matrix(y_test, rf_pred)

print("Random Forest Confusion Matrix:")
print(rf_cm)

Random Forest Confusion Matrix:
[[1215   10]
 [  41  492]]


## Conclusion

In this project, I built a machine learning model to predict whether a Netflix title is a Movie or a TV Show.

I selected relevant features from the dataset and encoded the categorical features using OneHotEncoder.

I trained two classification models: Logistic Regression and Random Forest.

Logistic Regression achieved an accuracy of 99.37%, while Random Forest achieved an accuracy of 97.10%.

Based on the test results, Logistic Regression performed better on this dataset.